RHIDP-5605  - Plugin installation documentation improvement for adding 'forceDownload' field (#988)
redhat-developer/red-hat-developers-documentation-rhdh: modules/dynamic-plugins/con-dynamic-plugins-cache.adoc

====

== Configuring the dynamic plugins cache

> You can set the following optional dynamic plugin cache parameters in your `dynamic-plugins.yaml` file:

Reviewer: Consider modularization.  "You can set" denotes a procedure module.

> You can set the following optional dynamic plugin cache parameters:

Reviewer 2: ```suggestion
You can set the following optional dynamic plugin cache parameters in your `dynamic-plugins.yaml` file:
```

If all of these parameters are being added to `dynamic-plugins.yaml`, then we can specify that here without having to do it for each individual parameter.

> * `forceDownload`: Set to `true` to force a reinstall of the plugin, bypassing the cache. Default is `false`. For example, modify your `dynamic-plugins.yaml` file as follows:

Reviewer 2: Is there supposed to be an example immediately after _as follows:_? I just see another bullet after this lead-in. :) 
EDIT: Oh, I see it further down, after the addition of the new parameters!

> * `forceDownload`: Set to `true` to force a reinstall of the plugin, bypassing the cache. Default is `false`. For example, modify your `dynamic-plugins.yaml` file as follows:

Reviewer 2: ```suggestion
* `forceDownload`: Set the value to `true` to force a reinstall of the plugin, bypassing the cache. The default value is `false`. 
```

Reviewer 2: Sometimes we start with a capital letter and sometimes with a lowercase. I'm not actually sure what the guidelines suggest, but we might want to use one style consistently

> * `pullPolicy`: this is similar to the `forceDownload` parameter and is consistent with other image container platforms. The possible settings for `pullPolicy` are:

Reviewer 2: ```suggestion
* `pullPolicy`: Similar to the `forceDownload` parameter and is consistent with other image container platforms. You can use one of the following values for this key:
```

** `Always`: compares the image digest in the remote registry and downloads the artifact if it has changed, even if the plugin was previously downloaded.
** `IfNotPresent`: downloads the artifact if it is not already present in the dynamic-plugins-root folder, without checking image digests.
+
[NOTE] 

> The `pullPolicy` setting is also applied to the NPM downloading method, although `Always` will download the remote artifact without a digest check. The existing `forceDownload` option remains functional, however,the `pullPolicy` option takes precedence. The `forceDownload` option may be deprecated in a future {product-short} release.

Reviewer 2: might vs. may

Author: Yep, I know the Vale linter suggests using 'might'...this seems more like 'might never happen' to me, 'may' is more intuitive to me as it is more than likely to happen. But let's keep that particular argument for a Style Guide discussion :-)

> For example, modify your dynamic-plugins.yaml file as follows to download the remote artifact without a digest check :

Reviewer 2: ```suggestion
.Example `dynamic-plugins.yaml` file configuration to download the remote artifact without a digest check:
```
